src/components/form/RadioButton.tsx

```diff
@@ -1,41 +1,35 @@
 import React, { useRef } from 'react';
 
-import { LegacyRadioButton as DesignSystemRadioButton } from '@digdir/design-system-react';
-import type { LegacyRadioButtonProps } from '@digdir/design-system-react';
+import { HelpText, Radio } from '@digdir/design-system-react';
+import type { RadioProps } from '@digdir/design-system-react';
 
 import classes from 'src/components/form/RadioButton.module.css';
 
-const Card = ({ children }: { children: React.ReactNode }) => <div className={classes.card}>{children}</div>;
-
-export interface IRadioButtonProps extends LegacyRadioButtonProps {
+export interface IRadioButtonProps extends Omit<RadioProps, 'children'> {
   showAsCard?: boolean;
+  label?: string;
+  helpText?: string;
+  hideLabel?: boolean;
 }
 
-export const RadioButton = ({ showAsCard = false, ...rest }: IRadioButtonProps) => {
+export const RadioButton = ({ showAsCard = false, label, helpText, hideLabel, ...rest }: IRadioButtonProps) => {
   const inputRef = useRef<HTMLInputElement>(null);
+  const Label = (
+    <div className={`${hideLabel ? 'sr-only' : ''} ${classes.radioLabelContainer}`}>
+      {label}
+      {helpText ? <HelpText title={helpText}>{helpText}</HelpText> : null}
+    </div>
+  );
   if (showAsCard) {
     return (
-      <Card>
-        {/** This element is only clickable for visual
-         effects. A screen reader would only want to click
-         the inner input element of the DesignSystemRadioButton. */}
-        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
-        <div
```

Comment:
Det å fjerne denne funksjonaliteten brekker nåværende oppførsel for radioknapper med `showAsCard`

Her er en screencast som viser hvordan disse radioknappene oppfører seg i dag:
![withcard](https://github.com/Altinn/app-frontend-react/assets/26043795/b47c9609-83f1-4173-8e62-7211af2058f8)

Og dette er oppførselen etter at denne koden fjernes:
![rec](https://github.com/Altinn/app-frontend-react/assets/26043795/e9703434-a3e4-47a1-9912-d49d18dfa763)

Det er snakk om å få et sånt option inn i selve designsystemet, men enn så lenge er dette kun implementert her i apps.

Reply:
Hei! Å jeg testet bare med label og ikke description i showAsCard og da ser jeg det skjer slik du viser, at man ikke kan klikke description for å velge. Det med at focus ringen ikke dukker opp er fordi den nye implementasjonen til Radioknappen ikke viser det heller med museklikk, men bare ved tastatur (de bruke `focus-visible`). Skal fikse det at man kan trykke på hele Card'et for å velge.

Reply:
Supert! 🙌 Ser også at det er blitt litt mindre padding på bunnen av kortet når det er med en description

Reply:
Ja så det jeg også, la det til også.